sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowPythonRunner.scala

```diff
@@ -87,6 +87,7 @@ class ArrowPythonRunner(
 
             arrowWriter.finish()
             writer.writeBatch()
+            dataOut.flush()
```

Comment:
This block of code will write to `dataOut` until it is blocked because the write buffer is full. I think adding a `flush` here will only hurt performance